### ISSUE TITLE
[SGO3] Direct firmware load for iwlwifi-cc-a0-78.ucode failed with error -2

### ISSUE THREAD

**Issue by FangCunWuChang** [linux-surface/linux-surface#1225]
WIFI firmware load error, so I can connect to WIFI through the command line, but I cannot connect from the GNOME Settings. It shows 'Wi-Fi adapter not found'.
And I am unable to connect to WIFI in the 5G frequency band.

### Environment

- Hardware model: Surface Go 3
- Kernel version: `Linux wuchang-surface 6.4.7-surface #3 SMP PREEMPT_DYNAMIC Tue Aug 1 21:23:55 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux`
- Distribution: Ubuntu 23.04

<details><summary>`dmesg` output</summary>

```
[5.131162] iwlwifi 0000:01:00.0: Direct firmware load for iwlwifi-cc-a0-78.ucode failed with error -2
[5.145208] iwlwifi 0000:01:00.0: api flags index 2 larger than supported by driver

```

</details>

**Comment by qzed** [comment]
That firmware doesn't seem to exist in the [linux-firmware git repo](https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/). I assume that since you can connect via the command line, the Gnome Settings issue is not a kernel driver issue. As for the 5GHz problem, I have no clue.

**Comment by FangCunWuChang** [comment]
I found the highest firmware API version supported of iwlwifi-cc is 78 in the kernel version 6.4 at here: <https://github.com/linux-surface/kernel/blob/37a244245e5a16b38a28238ef1e6215c7dccc1a4/drivers/net/wireless/intel/iwlwifi/cfg/22000.c#L13>  
So it will try to load `iwlwifi-cc-a0-78.ucode` first when kernel startup. This caused the error -2 (`ENOENT`). Then it will fall back and try to load `iwlwifi-cc-a0-77.ucode`.  
Actually, `iwlwifi-cc-a0-77.ucode` was successfully loaded:  

<details><summary>`dmesg` output</summary>

```
[5.498783] iwlwifi 0000:01:00.0: loaded firmware version 77.2dda880d.0 cc-a0-77.ucode op_mode iwlmvm
```

</details>

And `api flags index 2 larger than supported by driver` not reporting a real error, just the fact that the
firmware knows about more flags than the driver: <https://bugzilla.kernel.org/show_bug.cgi?id=212241#c3>  

-----

Why is the driver and firmware functioning normally, but the WiFi adapter was not found by GNOME. Now I am very confused.

**Comment by FangCunWuChang** [comment]
Now I found the STATE of `wlp1s0` in the result of `nmcli device` is `unmanaged`, but I had already set the `managed` flag to `true` in `/etc/NetworkManager/NetworkManager.conf`.  
Then I found this: <https://askubuntu.com/questions/1453042/nmcli-all-connections-show-unmanaged>  
> Networking on a server is usually managed by netplan, not Network Manager. In fact, NM is not installed on servers by default. Please edit your question to show the result of the terminal command: cat /etc/netplan/*.yaml  

And my system is Ubuntu Server 23.04. So I will try to change the management from `netplan` to `Network Manager`.
